### DISCUSSION TITLE
How to covert the renormalized force constant file xxx.xml at a finite temperature to the format xxx.fcs

### DISCUSSION THREAD

**Discussion by LiuHaoXuan0403** [ttadano/alamode#203] [Q&A] [answered]
Hi Dr. Tadano,
I try to use ALAMODE to compare the  renormalized force constant with the harmonic constant. Could you please give me some advice on how to convert the xxx.xml file to xxx.fcs file?

**Comment by LiuHaoXuan0403** [comment]
Hello Dr. Tadano
Could you please guide me what is the unit of FC2 in <HARMONIC> of xxx.xml. I want to compare the renormalized force constant at different temperature. Could you please give me some advice?
![image](https://github.com/ttadano/alamode/assets/49239192/28d5d970-0a52-4537-8000-b38c87d9d958)

**Reply by gangliu-github** [reply]
Hi, I also want to change xml file to FORCE_CONSTANTS file for phonopy. Do you clear the format of xml file?

**Reply by LiuHaoXuan0403** [reply]
The <HarmonicUnique> is same as XXX.fcs. But the pair1, pair2 and the unit in <HARMONIC> is not clear.  That's all I know

**Reply by gangliu-github** [reply]
Where can I found the format of xxx.fcs?

**Reply by ttadano** [reply, marked as answer]
The unit of force constants in *.xml is Ry/bohr^n for the n-th order force constant. The same unit is used also for the values in *.fcs.

Only irreducible sets of force constants are shown in *.fcs, where complete set of force constants are save in *.xml. To convert force constants in *.xml into the *.fcs format, we need to recover the information of force constant symmetry, so it would be quite tedious to achieve.

If you want to compare the difference between force constants in two different *.xml files, I recommend that you parse the values under `<HARMONIC>` using python, store them in a dictionary or Pandas dataframe, and then compare by taking a diff.

The meaning of `pair1` and `pair2` is a bit complicated:

- `pair1` = "`atom1` `xyz1`"

  `atom1` is the index of atom in a **primitive cell**, and `xyz1` labels the xyz component (1: x, 2: y, 3: z).

- `pair2` = "`atom2` `xyz2` `icell`"

  `atom2` is the index of atom in the **supercell**, and `xyz2` labels the xyz component.

  `icell` is additional index necessary to judge which periodic image of the supercell `atom2` is located. 
  Specifically, the `icell` is labelled as follows:
   1:  ( 0,  0,  0)
   2:  (-1, -1, -1)
   3:  (-1, -1,  0)
   4:  (-1, -1, +1)
   5:  (-1,  0,  -1)
        ....
  27: (+1, +1, +1)

**Reply by ttadano** [reply]
About converting the format into `FORCE_CONSTANTS` of phonopy, please see #192.

**Comment by LiuHaoXuan0403** [comment]
Hello, Tadano,

I get it. Thank you very much.

Best regards,
